examples/semi_supervised/plot_semi_supervised_newsgroups.py

```diff
@@ -3,17 +3,42 @@
 Semi-supervised Classification on a Text Dataset
 ================================================
 
-In this example, semi-supervised classifiers are trained on the 20 newsgroups
-dataset (which will be automatically downloaded).
+This example demonstrates the effectiveness of semi-supervised learning
+in text classification when labeled data is scarce.
+We compare four different approaches:
 
-You can adjust the number of categories by giving their names to the dataset
-loader or setting them to `None` to get all 20 of them.
+1. Supervised learning using 100% of labeled data (baseline)
 
+   - Uses SGDClassifier with TF-IDF features
+   - Represents the best possible performance with full supervision
+
+2. Supervised learning using only 20% of labeled data
```

Comment:
This on the other hand could be called a "baseline" to compare semi-supervised learning methods to.